src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -184,6 +180,12 @@ internal void Start()
                     throw Interop.GetExceptionForIoErrno(Interop.Sys.GetLastErrorInfo(), _fullDirectory, true);
                 }
 
+                // Normalize the _fullDirectory path to have a trailing slash
+                if (_fullDirectory[_fullDirectory.Length - 1] != '/')
```

Comment:
Ok.  Still makes me nervous.  Consider at least adding an assert for the length with a message to document the assumption that realpath will never return the empty string.